code/modules/mob/living/carbon/xenomorph/castes/carrier/abilities_carrier.dm

```diff
@@ -1,11 +1,26 @@
+#define LARVAL_HUGGER "larval hugger"
+#define CLAWED_HUGGER "clawed hugger"
+#define NEURO_HUGGER "neuro hugger"
+#define ACID_HUGGER "acid hugger"
+#define RESIN_HUGGER "resin hugger"
+
 //List of huggie types
-GLOBAL_LIST_INIT(hugger_type_list, typecacheof(list(
-		/obj/item/clothing/mask/facehugger/larval,
-		/obj/item/clothing/mask/facehugger/slash,
-		/obj/item/clothing/mask/facehugger/neuro,
-		/obj/item/clothing/mask/facehugger/acid,
-		/obj/item/clothing/mask/facehugger/resin,
-		)))
+GLOBAL_LIST_INIT(hugger_type_list, list(
+		LARVAL_HUGGER = /obj/item/clothing/mask/facehugger/larval,
+		CLAWED_HUGGER = /obj/item/clothing/mask/facehugger/slash,
+		NEURO_HUGGER = /obj/item/clothing/mask/facehugger/neuro,
+		ACID_HUGGER  = /obj/item/clothing/mask/facehugger/acid,
+		RESIN_HUGGER = /obj/item/clothing/mask/facehugger/resin,
+		))
+
+//List of huggie images
+GLOBAL_LIST_INIT(hugger_images_list,  list(
+		LARVAL_HUGGER = image('icons/mob/actions.dmi', icon_state = LARVAL_HUGGER),
+		CLAWED_HUGGER = image('icons/mob/actions.dmi', icon_state = CLAWED_HUGGER),
+		NEURO_HUGGER = image('icons/mob/actions.dmi', icon_state = NEURO_HUGGER ),
+		ACID_HUGGER  = image('icons/mob/actions.dmi', icon_state = ACID_HUGGER),
+		RESIN_HUGGER = image('icons/mob/actions.dmi', icon_state = RESIN_HUGGER),
+		))
```

Comment:
try doing this using less assoc possible, as assoc lists are heavier than normal lists or vars